libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1173,6 +1173,21 @@ void SMTEncoder::mergeVariables(set<VariableDeclaration const*> const& _variable
 		return var1->id() < var2->id();
 	};
 	set<VariableDeclaration const*, decltype(cmp)> sortedVars(begin(_variables), end(_variables), cmp);
+
+	/// Knowledge about references is erased if a reference is assigned,
+	/// so those also need their SSA's merged.
+	/// This thus not cause scope harm since the symbolic variables
```

Comment:
Do you mean `This does not cause scope harm`?

Reply:
Ah, yes. Updated.